### PR TITLE
[WIP][Do Not Review] Online payment with Braintree with 3DS2

### PR DESCRIPTION
**Pull request by HemaHG**
**TODOs**

- [x] verifying credit card details - extracted to [#2322](https://github.com/3scale/porta/pull/2322)
- [ ] issuing invoice
- [ ] verifying payment on purchase with braintree
- [ ] confirm payment on purchase with braintree on 3scale

**Which issue(s) this PR fixes**
fixes [THREESCALE-6495](https://issues.redhat.com/browse/THREESCALE-6495)

### CODE REVIEW

File: app/views/payment_gateways/_braintree_customer_form.html.erb [hunk 1]
```diff
@@ -188,11 +189,21 @@
                             console.error(tokenizeErr);
                             return;
                         }
-                        $('#braintree_nonce').val(payload['nonce']);
-                        $form.get(0).submit();
+                        var threeDSecure;
+                        braintree.threeDSecure.create({
+                            version: 2
```
Comment (damianpm) [inline app/views/payment_gateways/_braintree_customer_form.html.erb:194]:
```suggestion
                            version: 2,
```

Comment (damianpm) [reply app/views/payment_gateways/_braintree_customer_form.html.erb:194]: This typo is preventing the template to render ok

File: app/views/payment_gateways/_braintree_customer_form.html.erb [hunk 2]
```diff
@@ -127,15 +127,16 @@
...
-<script src="https://js.braintreegateway.com/web/3.46.0/js/client.min.js"></script>
-<script src="https://js.braintreegateway.com/web/3.46.0/js/hosted-fields.min.js"></script>
+<script src="https://js.braintreegateway.com/web/3.69.0/js/client.min.js"></script>
+<script src="https://js.braintreegateway.com/web/3.69.0/js/hosted-fields.min.js"></script>
+<script src="https://js.braintreegateway.com/web/3.69.0/js/three-d-secure.js"></script>
 <script type="text/javascript" charset="utf-8">
     var $form = $('#customer_form');
     var submit = document.querySelector('input[type="submit"]');
-
+    var client_token = @gateway_client.client_token
```
Comment (damianpm) [inline app/views/payment_gateways/_braintree_customer_form.html.erb:136]: You need to access ruby code using `<%= %>` tags

File: app/views/payment_gateways/_braintree_customer_form.html.erb [hunk 3]
```diff
@@ -129,6 +129,7 @@
 
 <script src="https://js.braintreegateway.com/web/3.69.0/js/client.min.js"></script>
 <script src="https://js.braintreegateway.com/web/3.69.0/js/hosted-fields.min.js"></script>
+<script src="https://js.braintreegateway.com/web/3.69.0/js/three-d-secure.js"></script>
```
Comment (hallelujah) [inline app/views/payment_gateways/_braintree_customer_form.html.erb:132]: Lets use the latest version 3.71 maybe? or put it in another PR?

**Review by hallelujah — COMMENTED** [review]
Just early review

The instructions for hosted fields are described here: https://developers.braintreepayments.com/guides/3d-secure/client-side/javascript/v3#hosted-fields

Maybe good to add it to the description too ;)

File: app/views/payment_gateways/_braintree_customer_form.html.erb (hunk 3 above)
Comment (Martouta) [reply app/views/payment_gateways/_braintree_customer_form.html.erb:132]: Better to extract this kind of risky changes to its own PR. In this case, this version is already in https://github.com/3scale/porta/pull/2310

Comment (HemaHG) [reply app/views/payment_gateways/_braintree_customer_form.html.erb:132]: upgraded to the latest version in [#2322](https://github.com/3scale/porta/pull/2322)

**Comment by damianpm** [comment]
Closed in favor of https://github.com/3scale/porta/pull/2322
